crates/gitbutler-branch-actions/tests/virtual_branches/list_details.rs

```diff
@@ -0,0 +1,113 @@
+use crate::virtual_branches::list;
+use gitbutler_branch_actions::BranchListingDetails;
+
+#[test]
+fn one_vbranch_on_integration_empty_details() -> anyhow::Result<()> {
+    let list = branch_details(
+        &list::project_ctx("one-vbranch-on-integration")?,
+        Some("virtual"),
+    )?;
+    assert_eq!(list.len(), 1);
+    assert_eq!(
+        list[0],
+        BranchListingDetails {
+            name: "virtual".into(),
+            lines_added: 0,
+            lines_removed: 0,
+            number_of_files: 0,
+            number_of_commits: 0,
+            authors: vec![],
+        }
+    );
+    Ok(())
+}
+
+#[test]
+fn one_vbranch_on_integration_single_commit() -> anyhow::Result<()> {
+    let list = branch_details(
+        &list::project_ctx("one-vbranch-on-integration-one-commit")?,
+        Some("virtual"),
+    )?;
+    assert_eq!(list.len(), 1);
+    assert_eq!(
+        list[0],
+        BranchListingDetails {
+            name: "virtual".into(),
+            lines_added: 2,
+            lines_removed: 0,
+            number_of_files: 2,
+            number_of_commits: 1,
+            authors: vec![default_author()],
+        }
+    );
+    Ok(())
+}
+
+#[test]
+fn many_commits_in_all_branch_types() -> anyhow::Result<()> {
+    let ctx = project_ctx("complex-repo")?;
+    let list = branch_details(&ctx, ["feature", "main", "non-virtual-feature"])?;
+    assert_eq!(list.len(), 3);
+    assert_eq!(
+        list[0],
+        BranchListingDetails {
+            name: "feature".into(),
+            lines_added: 100 + 5,
+            lines_removed: 0,
+            number_of_files: 1,
+            number_of_commits: 100 + 5, /* local tracking branch is merge base */
+            authors: vec![default_author()],
+        }
+    );
+    assert_eq!(
+        list[1],
+        BranchListingDetails {
+            name: "main".into(),
+            lines_added: 15,
+            lines_removed: 0,
+            number_of_files: 1,
+            // TODO(ST): why is it also going against the local tracking branch instead of the local `main`?
+            number_of_commits: 10 + 5,
+            authors: vec![default_author()],
+        }
+    );
+    assert_eq!(
+        list[2],
+        BranchListingDetails {
+            name: "non-virtual-feature".into(),
+            lines_added: 55,
+            lines_removed: 0,
+            number_of_files: 1,
+            number_of_commits: 50 + 5,
```

Comment:
This is essentially a merge-base test and it's a bit unexpected, as I was trying to make the merge-base logic observable. Sometimes, depending on order of operations, it creates virtual branches from git branches implicitly, even though here I managed to avoid that.

It's all about the local tracking branch `origin/main`, which is unchanged. I would have thought it's possible to make it diff against the local `main` though, just to see the logic in action - so far it always diffs against `origin/main`.